modules/reactor/test/tests/meshgenerators/reporting_id/cartesian_id/tests

```diff
@@ -0,0 +1,42 @@
+[Tests]
```

Comment:
Use requirement/detail description. These should also be more human readable; they should describe something that is achieved and not a copy pasta of an enum (i.e., something like assign_type=cell is bad)

Should probably be something like this:

```
[assembly]
  requirement = 'The system shall support the generation of pin reporting IDs for cartesian assemblies'
  [cell]
    ..
    detail = 'for each component in the lattice in sequential order'
  []
  [pattern]
    ..
    detail = 'based on the ID of the input tiles'
  []
  [manual]
    ..
    detail = 'based on user-defined mapping'
  []
[]
```